node-bindings/src/lib.rs

```diff
@@ -0,0 +1,605 @@
+    StacksBlockData, DEFAULT_DERIVATION_PATH,
+};
+use core::panic;
+use neon::prelude::*;
+use std::collections::BTreeMap;
+use std::path::PathBuf;
+use std::sync::mpsc;
+use std::thread;
+use std::{env, process};
+
+type DevnetCallback = Box<dyn FnOnce(&Channel) + Send>;
+
+struct StacksDevnet {
+    tx: mpsc::Sender<DevnetCommand>,
+    bitcoin_block_rx: mpsc::Receiver<BitcoinBlockData>,
+    stacks_block_rx: mpsc::Receiver<StacksBlockData>,
+    node_url: String,
+}
+
+enum DevnetCommand {
+    Start(Option<DevnetCallback>),
+    Stop(Option<DevnetCallback>),
+}
+
+impl Finalize for StacksDevnet {}
+
+impl StacksDevnet {
+    fn new<'a, C>(
+        cx: &mut C,
+        manifest_path: String,
+        logs_enabled: bool,
+        _accounts: BTreeMap<String, AccountConfig>,
+        devnet_overrides: DevnetConfigFile,
+    ) -> Self
+    where
+        C: Context<'a>,
+    {
+        let (tx, rx) = mpsc::channel::<DevnetCommand>();
+        let (meta_tx, meta_rx) = mpsc::channel();
+        let (log_tx, _log_rx) = mpsc::channel();
+        let (bitcoin_block_tx, bitcoin_block_rx) = mpsc::channel();
+        let (stacks_block_tx, stacks_block_rx) = mpsc::channel();
+
+        let channel = cx.channel();
+
+        let manifest_path = get_manifest_path_or_exit(Some(manifest_path.into()));
+        let devnet = DevnetOrchestrator::new(manifest_path, Some(devnet_overrides));
+
+        let node_url = devnet.get_stacks_node_url();
+
+        thread::spawn(move || {
+            if let Ok(DevnetCommand::Start(callback)) = rx.recv() {
+                // Start devnet
+                let (devnet_events_rx, terminator_tx) =
+                    match integrate::run_devnet(devnet, Some(log_tx), false) {
+                        Ok((Some(devnet_events_rx), Some(terminator_tx))) => {
+                            (devnet_events_rx, terminator_tx)
+                        }
+                        _ => std::process::exit(1),
+                    };
+                meta_tx
+                    .send(devnet_events_rx)
+                    .expect("Unable to transmit event receiver");
+
+                if let Some(c) = callback {
+                    c(&channel);
+                }
+
+                // Start run loop
+                while let Ok(message) = rx.recv() {
+                    match message {
+                        DevnetCommand::Stop(callback) => {
+                            terminator_tx
+                                .send(true)
+                                .expect("Unable to terminate Devnet");
+                            if let Some(c) = callback {
+                                c(&channel);
+                            }
+                            break;
+                        }
+                        DevnetCommand::Start(_) => break,
+                    }
+                }
+            } else {
+                // todo(ludo): Graceful termination.
```

Comment:
Maybe we should have some logging here for now until the graceful termination is in place.

Reply:
I removed that branch in https://github.com/hirosystems/clarinet/pull/157/commits/4a74c51f592ac68fd6948d7818fa5ff18f9eb7d2